align/schema/constraint.py

```diff
@@ -552,6 +552,103 @@ class MatchBlocks(SoftConstraint):
     instances: List[str]
 
 
+class PowerPorts(SoftConstraint):
+    '''
+    power port for each hieararchy
+    '''
+    ports: List[str]
+
+
+class GroundPorts(SoftConstraint):
+    '''
+    Ground port for each hieararchy
+    '''
+    ports: List[str]
+
+
+class ClockPorts(SoftConstraint):
+    '''
+    Clock port for each hieararchy
+    '''
+    ports: List[str]
+
+
+class DoNotUseLib(SoftConstraint):
+    '''
+    Primitive libraries which should not be used
+    '''
+    libraries: List[str]
+    propagate : Optional[bool]
+
+
+class IsDigital(SoftConstraint):
+    '''
+    Place this block digitally
+    Forbids any preprocessing, auto-annotation, array-identification or auto-constraint generation
+    '''
+    isTrue: bool
```

Comment:
Is IsTrue needed? Doesn't presence / absence of this constraint already indicate whether it is digital / not respectively? Same comment for all the other constraints.

Going forward, it might make sense to name the passes and then have a single constraint like "EnablePass" or "DisablePass" to turn them on / off. That way we are not expanding the constraint list for every pass we introduce

Having said that, this is an amazing start. I would want to see this merged as soon as possible if CI passes. And would like to see some testing as to what happens if some of the passes are actually disabled.

Reply:
That's a good suggestion. I will do it in separate PR.